src/vivarium/component.py

```diff
@@ -621,8 +624,7 @@ def build_lookup_table(
         self,
         builder: "Builder",
         # todo: replace with LookupTableData
-        data_source: Union[str, float, int, pd.DataFrame],
-        # todo: MIC-5029 remove argument when this is implemented
+        data_source: Union[str, float, int, list, pd.DataFrame],
```

Comment:
nit: add internal types to the list. (might need typing.List for python<3.9...I'm still confused by what is supported where)